server/src/test/java/org/elasticsearch/index/IndexServiceTests.java

```diff
@@ -0,0 +1,480 @@
+        // refresh every millisecond
+        execute("insert into test (x, data) values (2, 'foo')");
+        client().admin().indices().prepareUpdateSettings(indexName)
+            .setSettings(Settings.builder().put(IndexSettings.INDEX_REFRESH_INTERVAL_SETTING.getKey(), "1ms")).get();
+        assertTrue(refreshTask.isClosed());
+
+        assertBusy(() -> {
+            // this one becomes visible due to the force refresh we are running on updateMetadata if the interval changes
+            try (Engine.Searcher searcher = shard.acquireSearcher(indexName)) {
+                TopDocs search = searcher.search(new MatchAllDocsQuery(), 10);
+                assertEquals(2, search.totalHits.value);
+            }
+        });
+        execute("insert into test (x, data) values (3, 'foo')");
+
+        assertBusy(() -> {
+            // this one becomes visible due to the scheduled refresh
+            try (Engine.Searcher searcher = shard.acquireSearcher("test")) {
+                TopDocs search = searcher.search(new MatchAllDocsQuery(), 10);
+                assertEquals(3, search.totalHits.value);
+            }
+        });
+    }
+
+    public void testAsyncFsyncActuallyWorks() throws Exception {
+        execute("create table test(x int, data string) clustered into 1 shards with (\"translog.sync_interval\" = '100ms', " +
+                "\"translog.durability\" = 'ASYNC')");
+        IndexService indexService = getIndexService("test");
+        var indexName = indexService.index().getName();
+        ensureGreen(indexName);
+        assertTrue(indexService.getRefreshTask().mustReschedule());
+        execute("insert into test (x, data) values (1, 'foo')");
+        IndexShard shard = indexService.getShard(0);
+        assertBusy(() -> assertFalse(shard.isSyncNeeded()));
+    }
+
+    public void testRescheduleAsyncFsync() throws Exception {
+        execute("create table test(x int, data string) clustered into 1 shards with (\"translog.sync_interval\" = '100ms', \"translog.durability\" = 'REQUEST')");
+        IndexService indexService = getIndexService("test");
+        var indexName = indexService.index().getName();
+
+        ensureGreen(indexName);
+        assertNull(indexService.getFsyncTask());
+
+        execute("alter table test set (\"translog.durability\" = 'ASYNC')");
+
+        assertNotNull(indexService.getFsyncTask());
+        assertTrue(indexService.getFsyncTask().mustReschedule());
+        execute("insert into test (x, data) values (1, 'foo')");
+        assertNotNull(indexService.getFsyncTask());
+        final IndexShard shard = indexService.getShard(0);
+        assertBusy(() -> assertFalse(shard.isSyncNeeded()));
+
+        execute("alter table test set (\"translog.durability\" = 'REQUEST')");
+        assertNull(indexService.getFsyncTask());
+
+        execute("alter table test set (\"translog.durability\" = 'ASYNC')");
+        assertNotNull(indexService.getFsyncTask());
+    }
+
+    @Ignore("Flaky")
```

Comment:
Last remaining flaky test. Looking into it right now, we are a bit behind with translog related bp. I would that fix that in a follow up.